drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/enrollment_attribute_calculations.rb

```diff
@@ -0,0 +1,167 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module AllNeighborsSystemDashboard
+  module EnrollmentAttributeCalculations
+    extend ActiveSupport::Concern
+
+    included do
+      # Maybe update Tableau to use HUD event numbers?
+      SERVICE_CODE_ID = {
+        1 => 433, # Referral to Prevention Assistance project
+        # 2 => # Problem Solving/Diversion/Rapid Resolution intervention or service
+        3 => 435, # Referral to scheduled Coordinated Entry Crisis Needs Assessment
+        4 => 436, # Referral to scheduled Coordinated Entry Housing Needs Assessment
+        5 => 437, # Referral to Post-placement/ follow-up case management
+        6 => 438, # Referral to Street Outreach project or services
+        7 => 439, # Referral to Housing Navigation project or services
+        8 => 440, # Referral to Non-continuum services: Ineligible for continuum services
+        9 => 441, # Referral to Non-continuum services: No availability in continuum services
+        # 10 => # Referral to Emergency Shelter bed opening
+        11 => 443, # Referral to Transitional Housing bed/unit opening
+        12 => 444, # Referral to Joint TH-RRH project/unit/resource opening
+        13 => 445, # Referral to RRH project resource opening
+        14 => 446, # Referral to PSH project resource opening
+        15 => 447, # Referral to Other PH project/unit/resource opening
+        # 16 => 'Referral to emergency assistance/flex fund/furniture assistance',
+        17 => 1114, # Referral to Emergency Housing Voucher (EHV)
+        # 18 => # Referral to a Housing Stability Voucher
+      }.freeze
+
+      def household_type(enrollment)
+        return 'Children Only' if enrollment.children_only?
+        return 'Adults and Children' if enrollment.other_clients_under_18.positive? && enrollment.age > 17
+
+        'Adults Only'
+      end
+
+      def prior_living_situation_category(enrollment)
+        case enrollment.living_situation
+        when 30, 17, 37, 8, 9, 99
+          'Unknown'
+        when 16
+          'Unsheltered'
+        when 1, 18, 2
+          'Sheltered'
+        when 15, 6, 7, 25, 4, 5
+          'Institutional'
+        when 29, 14, 32, 13, 27, 12, 22, 35, 36, 23, 26, 28, 19, 3, 31, 33, 34, 10, 20, 21, 11
+          'Housed'
+        when 24
+          'Deceased'
+        else
+          'ERROR'
+        end
+      end
+
+      def exit_date(filter, enrollment)
+        date = enrollment.exit_date
+        return nil if date.present? && date > filter.end_date
+
+        date
+      end
+
+      def adjusted_edit_date(filter, enrollment)
+        date = enrollment.exit_date
+        return filter.end_date if date.blank?
+        return nil if date.present? && date > filter.end_date
```

Comment:
So, if we haven't exited, use the end of the report, but if we exited after the report end, ignore it?  That seems odd.

Reply:
It is odd, but here is the existing logic:
```sql
    ISNULL(
        CASE
            WHEN E.ExitDate > @ReportEnd
                THEN NULL
            ELSE E.ExitDate
        END,
        @ReportEnd
    ) AS [Adjusted Exit Date]
```

Reply:
I read that as, if the exit date is after report end, use report end, otherwise use the exit date.
IsNull is going to use @ReportEnd if the first item is null, which will be the case when the exit date is either NULL (no exit) or after the end date.  This should be the same as:
```
[
  enrollment.exit_date,
  filter.end_date,
].compact.min
```

Reply:
Ah -- so you read `ISNULL` as being the same as `COALESCE` -- makes sense... I read it as return null if any nulls.